### PR TITLE
SNOW-179637 Added support for parsing BigDecimal in AVRO converter

### PR DESCRIPTION
**Pull request by sfc-gh-zli**
Added support for parsing BigDecimal in AVRO converter

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/records/RecordService.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
   public static final SimpleDateFormat ISO_DATE_FORMAT= new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
   public static final SimpleDateFormat TIME_FORMAT= new SimpleDateFormat("HH:mm:ss.SSSZ");
-  private static final int MAX_SNOWFLAKE_NUMBER_PRECISION = 38;
+  public static final int MAX_SNOWFLAKE_NUMBER_PRECISION = 38;
```
Comment (sfc-gh-zli) [inline src/main/java/com/snowflake/kafka/connector/records/RecordService.java:57]: Moving this parameter to public for test to access.

Comment (sfc-gh-bli) [reply src/main/java/com/snowflake/kafka/connector/records/RecordService.java:57]: package private?

**Review by sfc-gh-bli — APPROVED** [review]

File: src/main/java/com/snowflake/kafka/connector/records/RecordService.java (hunk 1 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/records/RecordService.java]: ack

File: src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java [hunk 2]
```diff
@@ -126,6 +137,37 @@ public void testAvroWithSchemaRegistry() throws IOException
...
+    converter.setSchemaRegistry(schemaRegistry);
+
+    AvroConverter avroConverter = new AvroConverter(schemaRegistry);
+    avroConverter.configure(Collections.singletonMap("schema.registry.url", "http://fake-url"), false);
+    byte[] converted = avroConverter.fromConnectData(TEST_TOPIC, original.schema(), original);
+    SchemaAndValue avroInputValue = converter.toConnectData(TEST_TOPIC, converted);
+
+    SnowflakeRecordContent content = (SnowflakeRecordContent) avroInputValue.value();
+    assert content.getData()[0].toString()
+      .equals("{\"bytes\":\"foo\",\"bytesReadOnly\":\"foo\",\"bytesHex\":\"\\u0000¬\\u0000 ¿\",\"bytesDecimal\":1.1111111111111112E36}");
```
Comment (sfc-gh-zli) [inline src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:168]: byteHex will be ISO_8859_1 encoded and character escaped

File: src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java [hunk 3]
```diff
@@ -129,6 +138,66 @@ public void testAvroWithSchemaRegistry() throws IOException
...
+      .addToSchema(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.BYTES));
+
+    org.apache.avro.Schema schemaWithDecimal = org.apache.avro.SchemaBuilder
+      .record("MyRecord").fields()
+      .name("bytes").type().bytesType().noDefault()
+      .name("bytesReadOnly").type().bytesType().noDefault()
+      .name("bytesHex").type().bytesType().noDefault()
+      .name("bytesDecimal").type(decimalType).noDefault()
+      .endRecord();
+    System.out.println(schemaWithDecimal.toString(true));
```
Comment (sfc-gh-zli) [inline src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:155]: Schema looks like this: 
```
{
  "type" : "record",
  "name" : "MyRecord",
  "fields" : [ {
    "name" : "bytes",
    "type" : "bytes"
  }, {
    "name" : "bytesReadOnly",
    "type" : "bytes"
  }, {
    "name" : "bytesHex",
    "type" : "bytes"
  }, {
    "name" : "bytesDecimal",
    "type" : {
      "type" : "bytes",
      "logicalType" : "decimal",
      "precision" : 4,
      "scale" : 3
    }
  } ]
}
```

**Comment by chapm250** [comment]
Would this not being merged cause a 19, 4 decimal in avro to become bytes in snowflake?

**Comment by sfc-gh-zli** [comment]
@chapm250 Correct. Without this PR, BigDecimal are not going to be handled correctly.

**Pull request merged**